services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -425,83 +425,131 @@ public RoutesToNeutronWebResponse removeRoutesToNeutronRouter(String routerid, R
     }
 
     @Override
-    public ConnectedSubnetsWebResponse getConnectedSubnets(String projectId, String vpcId, String subnetId) throws ResourceNotFoundException, ResourcePersistenceException, SubnetNotBindUniquePortId {
+    public ConnectedSubnetsWebResponse getConnectedSubnets(String projectId, String vpcId, String subnetId) throws ResourceNotFoundException, ResourcePersistenceException, SubnetNotBindUniquePortId, UnsupportedOperationException {
         List<String> subnetIds = new ArrayList<>();
-        ConnectedSubnetsWebResponse connectedSubnetsWebResponse = new ConnectedSubnetsWebResponse();
+        InternalRouterInfo internalRouterInfo = new InternalRouterInfo();
+        ConnectedSubnetsWebResponse connectedSubnetsWebResponse = new ConnectedSubnetsWebResponse(internalRouterInfo, subnetIds);
+
+        SubnetWebJson subnetWebJson = this.routerToSubnetService.getSubnet(projectId, subnetId);
+        if (!this.ValidateSubnetAndConnectedRouter(subnetWebJson, subnetId)) {
+            logger.log(Level.WARNING, "Validation failed for SubnetAndConnectedRouter | subnet id:" + subnetId);
+            return connectedSubnetsWebResponse;
+        }
+
+        Router router = this.routerDatabaseService.getByRouterId(subnetWebJson.getSubnet().getAttachedRouterId());
+        if (!this.ValidateRouter(router)) {
+            logger.log(Level.WARNING, "Validation failed for router | router :" + router);
+            return connectedSubnetsWebResponse;
+        }
+
+        String routeTableType = router.getNeutronRouteTable().getRouteTableType();
+        Map<String, String> gwPortToSubnetIdMap = new HashMap<>();
+        if (routeTableType.equals("neutron_router")) {
```

Comment:
Should we take "neutron_router" as the constants format value?

Reply:
Yes. This was a bug found through debugging.